### ISSUE TITLE
Provider demands tag optional property to be defined as `[""]`

### ISSUE THREAD

**Issue by dlouseiro** [Snowflake-Labs/terraform-provider-snowflake#1394]
**Provider Version**

`0.51.0`

**Terraform Version**

`v1.3.6`

**Describe the bug**

When a `snowflake_tag` resource is created without `allowed_values`, the provider detects changes upon every `terraform plan/apply`, assuming that we're trying to change the value of `allowed_values` from `[""]` to `null`.

After some deeper investigation, I noticed that what is materialized in the `terraform` state when `allowed_values` is omitted is something like this:

```
"allowed_values": [
              ""
            ]
```

Although, when I run a `SHOW TAGS`, the allowed values are actually materialized as `NULL` when the `allowed_values` property is omitted, so if I run `terraform apply` multiple times, terraform always considers that the value for `allowed_values` is being changes from `null` to `[""]`.

In order to ensure state consistency, I needed to force `allowed_values = [""]` in my terraform configuration.

**Expected behavior**

I'd expect the omission of an optional property to produce consistent states between what is in the state file and what is materialized in the database.

**Comment by bogdan-cioata** [comment]
Same here, thanks reporting!

**Comment by bogdan-cioata** [comment]
As a workaround, until the bug is fixed, you can use instead:
```
lifecycle {
    ignore_changes = [
      allowed_values
    ]
  }
```

**Comment by dlouseiro** [comment]
> As a workaround, until the bug is fixed, you can use instead:
> 
> lifecycle {
>     ignore_changes = [
>       allowed_values
>     ]
>   }

Won't this also ignore changes in `allowed_values` when there are "actual" changes, as in, when we actually change the allowed values of this tag?

**Comment by bogdan-cioata** [comment]
the workaround is for when you don't fill `allowed_values`.

**Comment by dlouseiro** [comment]
I just upgraded the provider to version 0.61 and it seems that this problem is now fixed.

**Comment by sfc-gh-asawicki** [comment]
Hey @dlouseiro, can we close the issue then?

**Comment by sfc-gh-jmichalak** [comment]
Hi @dlouseiro @bogdan-cioata 👋 

This was already fixed in v0.61, but we've just released v0.99.0 ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.99.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0980--v0990)) in which we reworked tags. Please upgrade to this version and let us know if you have any issues.

**Comment by sfc-gh-dszmolka** [comment]
closing based on the comments. let us know please if you still see any issues with the latest provider (v1)
